capabilityreceiver/cmd/migratevalidations/main_integration_test.go

```diff
@@ -0,0 +1,272 @@
+	th.Assert(t, err == nil, fmt.Sprintf("Error while adding first metadata object: %s", err))
+	metadataID2, err := store.AddFHIREndpointMetadata(ctx, &testMetadata2)
+	th.Assert(t, err == nil, fmt.Sprintf("Error while adding first metadata object: %s", err))
+
+	// Put two FHIR endpoints in the history table
+	tlsVersion := "1.3"
+	mimeTypes := []string{"application/json+fhir"}
+	firstTime := time.Now().UTC().Round(time.Microsecond)
+	url1 := "www.testurl.com/cerner/DSTU2"
+	_, err = store.DB.ExecContext(ctx, addFHIREndpointInfoStatement, url1, "I", capStat1, tlsVersion, pq.Array(mimeTypes), metadataID1, firstTime)
+	th.Assert(t, err == nil, fmt.Sprintf("Error when adding to the database %s", err))
+
+	secondTime := time.Now().UTC().Round(time.Microsecond)
+	url2 := "www.testurl.com/epic/DSTU2"
+	_, err = store.DB.ExecContext(ctx, addFHIREndpointInfoStatement, url2, "I", capStat2, tlsVersion, pq.Array(mimeTypes), metadataID2, secondTime)
+	th.Assert(t, err == nil, fmt.Sprintf("Error when adding to the database again %s", err))
+
+	resultCh := make(chan Result)
+
+	// Check that data only updates the first URL
+	defaultArgs := make(map[string]interface{})
+	defaultArgs["historyArgs"] = historyArgs{
+		fhirURL:   url1,
+		store:     store,
+		result:    resultCh,
+		isHistory: true,
+	}
+
+	go addToValidationTable(ctx, &defaultArgs)
+	for res := range resultCh {
+		th.Assert(t, res.URL == url1, fmt.Sprintf("Returned result URL is not equal to %s, is instead %s", url1, res.URL))
+		close(resultCh)
+	}
+
+	historyRows, err := store.DB.QueryContext(ctx, getFHIREndpointInfoStatement, url1)
+	th.Assert(t, err == nil, fmt.Sprintf("error getting data from fhir_endpoints_info: %s", err))
+	// Check to make sure exactly 1 ID was updated
+	count := 0
+	var valID int
+	for historyRows.Next() {
+		th.Assert(t, count < 1, fmt.Sprintf("should only be one item in the database for this URL"))
+		var receivedTime time.Time
+		err = historyRows.Scan(&receivedTime, &valID)
+		th.Assert(t, err == nil, fmt.Sprintf("Error while scanning the rows of the history table for URL %s. Error: %s", url1, err))
+		th.Assert(t, receivedTime.Equal(firstTime), fmt.Sprintf("The time was updated to %+v from %+v", receivedTime, firstTime))
+		count++
+	}
+	th.Assert(t, count == 1, "should be one item in the database, instead is 0")
+
+	// Make sure that ID was added to the validation results table
+	valResRows, err := store.DB.QueryContext(ctx, getValidationResultStatement, valID)
+	th.Assert(t, err == nil, fmt.Sprintf("error getting data from validation_results: %s", err))
+	defer valResRows.Close()
+
+	for valResRows.Next() {
+		var valResCount int
+		err = valResRows.Scan(&valResCount)
+		th.Assert(t, valResCount == 1, fmt.Sprintf("for URL %s, there should be one row with id %d", url1, valID))
+		count++
+	}
+
+	// Make sure that 5 entries were added to the validation table with that ID
+	valRows, err := store.DB.QueryContext(ctx, getValidationStatement, valID)
```

Comment:
Here you could use probably QueryRowContext to directly scan the count rather than having to call valRows.next(), since this query should only return one row anyways